src/main/java/org/mate/bookstorespringboot/service/impl/UserDetailsServiceImpl.java

```diff
@@ -0,0 +1,21 @@
+package org.mate.bookstorespringboot.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import org.mate.bookstorespringboot.repository.UserRepository;
+import org.mate.bookstorespringboot.service.interfaces.UserDetailsService;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserDetailsServiceImpl implements UserDetailsService {
```

Comment:
You have the same logic in CustomUserDetailsService. Do you really need both classes?